src/cromshell/submit/command.py

```diff
@@ -0,0 +1,131 @@
+    io_utils.assert_required_file_is_not_empty(dependencies_zip, "Dependencies Zip")
+
+    # At this point, we should validate our inputs if we can:
+    womtool_validate_wdl_and_json(wdl, wdl_json)
+
+    # Check if Cromwell Server Backend works
+    http_utils.assert_can_communicate_with_server(config)
+
+    # Submit workflow to Cromwell Server
+    LOGGER.info(f"Submitting job to server: {cromshellconfig.cromwell_server}")
+    submission_params = {
+        'workflowSource': open(wdl, 'rb'),
+        'workflowInputs': open(wdl_json, 'rb')
+    }
+    if options_json is not None:
+        submission_params['workflowOptions'] = open(options_json, 'rb')
+    if dependencies_zip is not None:
+        submission_params['workflowDependencies'] = open(dependencies_zip, 'rb')
+
+    # TODO: In order for this to work the ending "/" needs be removed from config.api_string
+    # TODO: This might need to be implemented in another PR because it involves editing
+    # TODO: the status command and http_util script
+    request_out = requests.post(f"{config.cromwell_server}{config.api_string}",
+                                files=submission_params)
+
+    # Check to make sure that we actually submitted the job correctly
+    # 1. Check for any initial failure by server to accept the job.
+    http_utils.check_http_request_status_code(request_out)
+
+    # 2. Check messages from server for workflow problems.
+    # Get our workflow status and job ID:
+    workflow_status = json.loads(request_out.content)
+
+    # 2.A If the status is not `Submitted`, something went wrong: Todo: did a manual test but not one with failed workflow
+    if workflow_status['status'] != "Submitted":
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Server reports that job was not properly submitted.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Server reports that job was not properly submitted.\n"
+            f"Server response:{request_out.text}")
+
+    # 2.B If the ID is not an ID, something went wrong:
+    if io_utils.validate_workflow_id(workflow_status['id']) != 0:
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Did not get a valid ID back. Something went wrong.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Did not get a valid ID back. Something went wrong.\n"
+            f"Server response:{request_out.text}")
+
+    # If we get here, we successfully submitted the job and should track it locally:
+    log.display_logo(io_utils.turtle)
+    io_utils.pretty_print_json(request_out.text)
+
+    # Create a directory to hold function input files, using server name
+    run_directory = Path(config.config_dir)/config.folder_url/workflow_status['id']
+    Path.mkdir(Path(run_directory), parents=True, exist_ok=True)  # should it be ok that it exists?
```

Comment:
I'm not understanding the comment here.
Is it a question about what's the appropriate thing to do if the run directory already exists?
If so, I think it's OK if that directory exists.

Reply:
Thats right. I doubt the dir would already exist because the workflow_id is being used to create the dir name. 
But for whatever reason if it did exist should it be allowed to continue, and overwrite the contents of the folder with the new wdl, wdl_input_json in the next few lines. Or should it fail here and not copy over any files to the existing folder.

Reply:
That scenario would be rare.
My personal opinion would be to error out on 2nd thought, as IDs should be unique, which is an assumption that seems to be violated if the dir exists.
Aliasing a run is a different story though, since I could see why people may want to re-use some aliases sometimes.

Reply:
sounds good, added exception handling to log the event